src/messaging/ExchangeMgr.cpp

```diff
@@ -109,6 +117,9 @@ CHIP_ERROR ExchangeManager::Shutdown()
     }
 
     mState = State::kState_NotInitialized;
+#if CONFIG_DEVICE_LAYER
+    DeviceLayer::PlatformMgr().UnlockChipStack();
```

Comment:
Same as above, move DeviceLayer::PlatformMgr().UnlockChipStack(); to the place Shutdown is called from the client thread